lightning/src/ln/channelmanager.rs

```diff
@@ -10366,8 +10366,17 @@ where
 				}
 			},
 			OffersMessage::Invoice(invoice) => {
+				let expected_payment_id = match context {
+					OffersContext::OutboundPayment { payment_id } => payment_id,
+					_ => return ResponseInstruction::NoResponse,
+				};
+
 				let result = match invoice.verify(expanded_key, secp_ctx) {
 					Ok(payment_id) => {
+						if payment_id != expected_payment_id {
+							return ResponseInstruction::NoResponse;
+						}
```

Comment:
@TheBlueMatt If we receive an invoice for a refund _without_ blinded paths, can we actually authenticate it? With this change, we'd see `OffersContext::Unknown` and so would return `ResponseInstruction::NoResponse`.

But if we skip the check in that case, someone could send us an invoice for a refund using an unrelated blinded path created before this PR (i.e., one where we'd get `OffersContext::Unknown` here). While this wouldn't de-anonymize us -- the refund contains our pubkey -- it could tie us to an old offer that uses a transient pubkey.

Reply:
I suppose this is also the case for an offer without blinded paths. Although in that case we *can* authenticate, but it could tie us to an older offer that uses a transient pubkey.